plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/shared.ts

```diff
@@ -325,3 +326,51 @@ export const initProductScreenTracks = () => {
 			}
 		} );
 };
+
+export function addExitPageListener( pageId: string ) {
+	let productChanged = false;
+	let triggeredDelete = false;
+
+	const deleteButton = document.querySelector( '#submitpost a.submitdelete' );
+
+	if ( deleteButton ) {
+		deleteButton.addEventListener( 'click', function () {
+			triggeredDelete = true;
+		} );
+	}
+
+	function checkIfSubmitButtonsDisabled() {
+		const submitButtonSelectors = [
+			'#submitpost [type="submit"]',
+			'#submitpost #post-preview',
+		];
+		let isDisabled = false;
+		for ( const sel of submitButtonSelectors ) {
+			document.querySelectorAll( sel ).forEach( ( element ) => {
+				if ( element.classList.contains( 'disabled' ) ) {
+					isDisabled = true;
+				}
+			} );
+		}
+		return isDisabled;
+	}
+	window.addEventListener( 'beforeunload', function ( event ) {
+		// Check if button disabled or triggered delete to see if user saved or deleted the product instead.
+		if ( checkIfSubmitButtonsDisabled() || triggeredDelete ) {
```

Comment:
As the comment implies, it was hard to differentiate if the `beforeunload` was happening because a user was saving or deleting a product or if they were exiting.
These two checks were the main differentiator I could add to check against this.